crates/uv-workspace/src/workspace.rs

```diff
@@ -0,0 +1,69 @@
+use std::path::{Path, PathBuf};
+
+use uv_fs::Simplified;
+
+use crate::{Options, PyProjectToml};
+
+/// Represents a project workspace that contains a set of options and a root path.
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct Workspace {
+    options: Options,
+    root: PathBuf,
+}
+
+impl Workspace {
+    /// Find the [`Workspace`] for the given path.
+    ///
+    /// The search starts at the given path and goes up the directory tree until a workspace is
+    /// found.
+    pub fn find(path: impl AsRef<Path>) -> Result<Option<Self>, WorkspaceError> {
+        for ancestor in path.as_ref().ancestors() {
+            if let Some(options) = read_options(ancestor)? {
```

Comment:
Does this mean we only pick the closest file?

The scenario i'm thinking about is one where the work dir is `/home/ferris/foo/bar/` and we have:
* `/home/ferris/uv.toml` or `/home/ferris/.config/uv.toml`: Has some general user-level settings
* `/home/ferris/foo/pyproject.toml`: Defines a workspace of python projects and has some project level settings
* `/home/ferris/foo/bar/pyproject.toml`: A member of that workspace. Does not contain settings (i think?)

In this case, i want to first load user-level setting, then override them with the project level settings (implicitly), then resolve `bar` as part of the `foo` workspace

Reply:
Yeah we only pick the closest file. What if we do the same as Ruff, and skip a `pyproject.toml` that lacks a `tool.uv` entry?

Reply:
That'd make sense to me.

Reply:
What do you think about https://doc.rust-lang.org/cargo/reference/config.html#hierarchical-structure ?